quinn-proto/src/connection/mod.rs

```diff
@@ -2732,6 +2738,18 @@ where
         self.in_flight.bytes + u64::from(self.mtu) >= self.path.congestion.window()
     }
 
+    /// Whether UDP transmits are currently blocked by pacing
+    fn pacing_blocked(&mut self, now: Instant) -> bool {
+        let smoothed_rtt = self.path.rtt.conservative();
+        let window = self.path.congestion.window();
+        if let Some(delay) = self.path.pacing.delay(smoothed_rtt, self.mtu, window, now) {
+            self.timers.set(Timer::Pacing, delay);
```

Comment:
I don't really like the mixing of query and side effect in this method. Can we just outline the call to `self.timers.set()` into the `poll_transmit()` caller? I think that will make it clearer what's going on. The next question is then whether this method still carries its weight, maybe it should just be inlined into `poll_transmit()` altogether, since it's single-caller anyway.

Reply:
Good idea, thanks!